src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === typeof null) {
+        result += `${separator}`;
+        continue;
+      }
+
+      if (typeof this[i] === typeof undefined) {
+        result += `${separator}`;
+        continue;
+      }
```

Comment:
Уникай дублювання коду
```suggestion
      if (this[i] === null || this === undefined) {
        result += `${separator}`;
        continue;
      }
```